### PR TITLE
Currency fields #165205282

### PR DESCRIPTION
**Pull request by software-project**
https://www.pivotaltracker.com/story/show/165205282

### CODE REVIEW

**Review by akegan — CHANGES_REQUESTED** [review]
A couple of issues with QA:
1. Rental assistance amounts don't have currency field, don't load with $ format built-in
1. Lease values (rent, etc) don't load with $ format built in (should be able to add it in the lease model
1. Please update paper application to be consistent if it's not too much work

**Comment by software-project** [comment]
@akegan should be ready for a review

File: app/javascript/components/supplemental_application/SupplementalApplicationPage.js [hunk 1]
```diff
@@ -216,6 +217,8 @@ class SupplementalApplicationPage extends React.Component {
       rentalAssistance,
       this.state.persistedApplication.id
     )
+    // show price in right format
+    rentalAssistance.assistance_amount = formUtils.formatPrice(rentalAssistance.assistance_amount)
```
Comment (akegan) [inline app/javascript/components/supplemental_application/SupplementalApplicationPage.js:221]: @software-project I wonder if we could get this line covered with a snapshot test that verifies that currencies are formatted with $, etc.

File: spec/fixtures/model/force/lease/lease_domain.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
   "application_id": "a0o0P00000FFc2yQAD",
   "id": "a130U0000001n4wQAA",
   "lease_start_date": {"year": "2028", "month": "01", "day": "01"},
-  "monthly_parking_rent": 100,
-  "monthly_tenant_contribution": 800,
+  "monthly_parking_rent": "$100.00",
```
Comment (akegan) [inline spec/fixtures/model/force/lease/lease_domain.json]: maybe increase some of these sample currencies to >=$1,000.00 to catch the lack of commas issue.

**Review by akegan — CHANGES_REQUESTED** [review]
More QA notes:
- For some reason, when currencies are loaded for leases and rental assistances they are not formatted with commas (they do have the dollar sign and cents).
- @software-project should I move changing paper app currencies to currency field to a new ticket, or can you include in this PR?
- See comment below re: tests. 

Pic for currency formatting issue on load:
image

**Comment by akegan** [comment]
@software-project please also add review instructions to the ticket

File: app/javascript/components/supplemental_application/SupplementalApplicationPage.js (hunk 1 above)
Comment (software-project) [reply app/javascript/components/supplemental_application/SupplementalApplicationPage.js:221]: It's already covered in rental-assistance e2e test. I modified it just now to also test for format with comma.

**Comment by akegan** [comment]
getting weird behavior for non-number characters 
weirdCurrencyBehavior

Looks like in the [react final form example](https://codesandbox.io/s/3rp260ly51?from-embed)it also gets rid of letters if you type them in but doesn't have the odd behavior around commas

**Comment by akegan** [comment]
otherwise looks good! if the weird behavior from chars after commas is unavoidable I can check with Emily to see if we can keep it as is

**Comment by software-project** [comment]
@akegan should be working fine now.

**Review by akegan — APPROVED** [review]
OK looks good now. Thanks!

**Pull request merged**